### PR TITLE
feat: [#175800057,#176870138] Show when credit-card/bancomat/cobadge is expired

### PR DESCRIPTION
**Pull request by DavideValdo**
## Description
This PR adds an advice when credit card / bancomat / cobadge is expired

| credit card  | bancomat | cobadge |
| ------------- | ------------- | ------------- |
credit card | bancomat | cobadge

### other changes
- according to @thisisjp the icon size in `InfoBox` component is increased from 24 to 32 and its top margin is 4 as default

### how to test
- set your dev-server to this branch `card-expired` it will serve only expired cards

### CODE REVIEW

**Comment by thisisjp** [comment]
@DavideValdo my bad, the "delete this method" button at the bottom of the screen is part of this story :) 

+ could you please make the (!) icon same size as the other screens? You can check the InfoBox in UI Showroom as a benchmark.
+ there's some margin missing between the infobox and the CTA. Please check the prototype for the right specs

**Comment by DavideValdo** [comment]
@thisisjp https://github.com/pagopa/io-app/pull/2771 Should we wait for this to be merged then? I had done the button-related work but removed it as I realized it was another's PR matter

**Comment by thisisjp** [comment]
@DavideValdo yes, the story needs those 3 points to be fixed

**Comment by DavideValdo** [comment]
@thisisjp Hope it looks better like this, should I also remove the `+` icon next to the CTA that I put in the mix by mistake?

Schermata 2021-02-04 alle 17 33 31

**Comment by thisisjp** [comment]
@DavideValdo yes please, let's stick to the prototype 🙏 Is the font-size within the button same size as the other ones? It looks smaller—am I wrong? 🤔

**Comment by DavideValdo** [comment]
I actually copied the button from the services' list, I don't really see a different size. Maybe you're seeing a difference between the simulator's font antialiasing and the UI prototype.

File: ts/utils/dates.ts [hunk 1]
```diff
@@ -69,16 +69,10 @@ export function format(
...
-    today.getFullYear().toString().slice(2),
-    10
-  );
-  return (
-    expireYear < currentYear ||
-    (expireYear === currentYear && expireMonth < currentMonth)
-  );
+  // We pass the first day of the month, but endOfMonth() returns a Date object
+  // representing the last day of the target month.
+  const date = endOfMonth(`${expireMonth}/01/${expireYear}`);
```
Comment (Undermaken) [inline ts/utils/dates.ts:74]: Maybe this fixes also this bug https://www.pivotaltracker.com/story/show/176870138
Could you add some other test to check if it works both with year in 2 and 4 digits?
ex
`isExpired(3, 20)` should be the same of `isExpired(3, 2020)`

File: ts/utils/wallet.ts [hunk 2]
```diff
@@ -13,19 +12,11 @@ import { NumberFromString } from "./number";
...
-  // if we can't decode month or year value, card will be considered as expired
-  if (ccExpireYear === undefined || ccExpireMonth === undefined) {
-    return true;
-  }
-  return isExpired(ccExpireMonth, ccExpireYear);
+  const { expireYear, expireMonth } = creditCard;
+
+  // In case the date is not valid, this call will return -1
+  // thus considering the card as expired
+  return isExpired(Number(expireMonth), Number(expireYear));
```
Comment (Undermaken) [inline ts/utils/wallet.ts:19]: this function changes the previous behavior:

> // if we can't decode month or year value, card will be considered as expired

If you run this test

`expect(isExpired(Number("AAA"), Number("BBB"))).toBe(true);`

it fails

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: locales/it/index.yml [hunk 3]
```diff
@@ -1239,7 +1245,7 @@ saveCard:
 cardComponent:
   validUntil: Valida fino al
   expiresOn: Scade il
-  expiredCard: CARTA SCADUTA IL
+  expiredCard: Carta scaduta il
```
[the remark this answers is not in the document]
Comment (DavideValdo) [reply locales/it/index.yml:1248]: There are none, I double-checked

File: ts/features/wallet/bancomat/screen/BancomatDetailScreen.tsx [hunk 4]
```diff
@@ -85,6 +87,12 @@ const BancomatDetailScreen: React.FunctionComponent<Props> = props => {
     icon: pagoBancomatImage,
     caption: bancomat.abiInfo?.name ?? I18n.t("wallet.methods.bancomat.name")
   });
+
+  const { expireMonth, expireYear } = bancomat.info;
+  const cardIsExpired = bancomat
```
[the remark this answers is not in the document]
Comment (DavideValdo) [reply ts/features/wallet/bancomat/screen/BancomatDetailScreen.tsx:92]: Are you ok with still destructuring and take advantage of `CreditCardExpirationMonth.is` for strings in `isExpiredCard` and just a false check for `NaN` in `isExpired`?

File: locales/it/index.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavideValdo) [reply locales/it/index.yml:1277]: There are none, I double-checked

Screenshot 2021-02-10 at 21 18 42

**Pull request closed without merging**